plugins/vpp/aclplugin/data_resync_test.go

```diff
@@ -0,0 +1,419 @@
+		if sendControlPing {
+			sendControlPing = false
+			data := &vpe.ControlPingReply{}
+			reply, err := vppMock.ReplyBytes(request, data)
+			Expect(err).To(BeNil())
+			msgID, err := vppMock.GetMsgID(data.GetMessageName(), data.GetCrcString())
+			Expect(err).To(BeNil())
+			return reply, msgID, true
+		}
+
+		for _, dataMock := range dataList {
+			if request.MsgID == dataMock.Id {
+				// Send control ping next iteration if set
+				sendControlPing = dataMock.Ping
+				msgID, err := vppMock.GetMsgID(dataMock.Message.GetMessageName(), dataMock.Message.GetCrcString())
+				Expect(err).To(BeNil())
+				reply, err := vppMock.ReplyBytes(request, dataMock.Message)
+				Expect(err).To(BeNil())
+				return reply, msgID, true
+			}
+		}
+
+		replyMsg, msgID, ok := vppMock.ReplyFor(request.MsgName)
+
+		if ok {
+			reply, err := vppMock.ReplyBytes(request, replyMsg)
+			Expect(err).To(BeNil())
+			return reply, msgID, true
+		}
+
+		return reply, 0, false
+	}
+}
+
+var acls = []*acl.AccessLists_Acl{
+	{AclName: "acl1",
+		Rules: []*acl.AccessLists_Acl_Rule{
+			{
+				AclAction: acl.AclAction_PERMIT,
+				Match: &acl.AccessLists_Acl_Rule_Match{
+					IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+						Ip: &acl.AccessLists_Acl_Rule_Match_IpRule_Ip{
+							SourceNetwork:      "192.168.1.1/32",
+							DestinationNetwork: "10.20.0.1/24",
+						},
+					},
+				},
+			},
+		},
+		Interfaces: &acl.AccessLists_Acl_Interfaces{
+			Ingress: []string{"if1"},
+			Egress:  []string{"if2"},
+		},
+	},
+	{AclName: "acl2",
+		Rules: []*acl.AccessLists_Acl_Rule{
+			{
+				AclAction: acl.AclAction_PERMIT,
+				Match: &acl.AccessLists_Acl_Rule_Match{
+					MacipRule: &acl.AccessLists_Acl_Rule_Match_MacIpRule{
+						SourceAddress:        "192.168.0.1",
+						SourceAddressPrefix:  uint32(16),
+						SourceMacAddress:     "11:44:0A:B8:4A:35",
+						SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+					},
+				},
+			},
+		},
+		Interfaces: &acl.AccessLists_Acl_Interfaces{
+			Ingress: []string{"if3"},
+			Egress:  nil,
+		},
+	},
+}
+
+// Test synchronisation - writes ACLs to the empty VPP
+func TestResyncEmpty(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := aclTestSetup(t, false)
+	defer aclTestTeardown(connection, plugin)
+
+	ctx.MockVpp.RegisterBinAPITypes(acl_api.Types)
+	ctx.MockVpp.MockReplyHandler(vppMockHandler(ctx.MockVpp, []*vppReplyMock{
+		{
+			Id:   1011,
+			Ping: true,
+			Message: &acl_api.ACLDetails{},
+		},
+		{
+			Id:   1015,
+			Ping: true,
+			Message: &acl_api.ACLInterfaceListDetails{},
+		},
+		{
+			Id:   1013,
+			Ping: true,
+			Message: &acl_api.MacipACLDetails{},
+		},
+		{
+			Id:   1017,
+			Ping: true,
+			Message: &acl_api.MacipACLInterfaceListDetails{},
+		},
+		{
+			Id:   1001,
+			Ping: false,
+			Message: &acl_api.ACLAddReplaceReply{},
+		},
+		{
+			Id:   1005,
+			Ping: false,
+			Message: &acl_api.MacipACLAddReply{},
+		},
+	}))
+
+	err := plugin.Resync(acls)
+	Expect(err).To(BeNil())
```

Comment:
please create two new methods - for l2AclIndexes and l3l4AclIndexes - in acl configurator which allow to lookup in-memory mapping  ([example](https://github.com/ligato/vpp-agent/blob/master/plugins/defaultplugins/ifplugin/interface_config.go#L144-L147)) and check also mapping state in every test here (same as [here](https://github.com/ligato/vpp-agent/blob/master/plugins/defaultplugins/ifplugin/interface_config_test.go#L220-L222)). Plugin internal state is essential and it would be good if it could be tested also in acl_configurator_test